invoiceregistry.go

```diff
@@ -721,7 +608,47 @@ type registryHtlc struct {
 	resolve       func(HtlcResolution)
 }
 
+func (i *InvoiceRegistry) resolveViaDb(ctx context.Context,
+	h *registryHtlc) error {
+
+	dbInvoice, htlcs, err := i.cdb.Get(ctx, h.rHash)
+	if err != nil {
+		return err
+	}
+
+	i.logger.Debugw("Loaded settled invoice from db", "hash", h.rHash)
+
+	// Handle replays to a settled invoice.
+	if len(htlcs) == 0 {
+		return errors.New("unexpected unsettled invoice")
+	}
+
+	// If this htlc was used for settling the invoice,
+	// resolve to settled again.
+	if _, ok := htlcs[h.circuitKey]; ok {
+		h.resolve(NewSettleResolution(
+			dbInvoice.PaymentPreimage,
+			ResultReplayToSettled,
+		))
+
+		return nil
+	}
+
+	// Otherwise fail the htlc.
+	h.resolve(NewFailResolution(ResultInvoiceNotOpen))
+
+	return nil
+}
+
 func (i *InvoiceRegistry) process(ctx context.Context, h *registryHtlc) error {
+	logger := i.logger.With("hash", h.rHash)
+
+	// First try to resolve via the database, in case this is a replay.
+	err := i.resolveViaDb(ctx, h)
+	if err != types.ErrInvoiceNotFound {
```

Comment:
I think this error handling should be absorbed by `.resolveViaDb`

Reply:
Done, used additional `resolved` return value.